server/src/main/java/org/opensearch/common/cache/stats/CacheStatsResponse.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A class containing the 5 metrics tracked by a CacheStats object.
+ */
+public class CacheStatsResponse implements Writeable { // TODO: Make this extend ToXContent.
+    public CounterMetric hits;
+    public CounterMetric misses;
+    public CounterMetric evictions;
+    public CounterMetric sizeInBytes;
+    public CounterMetric entries;
+
+    public CacheStatsResponse(long hits, long misses, long evictions, long memorySize, long entries) {
+        this.hits = new CounterMetric();
+        this.hits.inc(hits);
+        this.misses = new CounterMetric();
+        this.misses.inc(misses);
+        this.evictions = new CounterMetric();
+        this.evictions.inc(evictions);
+        this.sizeInBytes = new CounterMetric();
+        this.sizeInBytes.inc(memorySize);
+        this.entries = new CounterMetric();
+        this.entries.inc(entries);
+    }
+
+    public CacheStatsResponse(StreamInput in) throws IOException {
+        this(in.readVLong(), in.readVLong(), in.readVLong(), in.readVLong(), in.readVLong());
+    }
+
+    public CacheStatsResponse() {
+        this(0, 0, 0, 0, 0);
+    }
+
+    public synchronized void add(CacheStatsResponse other) {
+        if (other == null) {
+            return;
+        }
+        this.hits.inc(other.hits.count());
+        this.misses.inc(other.misses.count());
+        this.evictions.inc(other.evictions.count());
+        this.sizeInBytes.inc(other.sizeInBytes.count());
+        this.entries.inc(other.entries.count());
+    }
```

Comment:
Is this the immutable representation or the mutable counter? We shouldn't use one class for both.

Reply:
Sure, this was the mutable counter, but we can add an immutable version.